Event comments CRUD

daniel-santana opened the pull request:
Hopefully
closes #40

daniel-santana committed:
added CRUD for events comments

lubien commented on app/Controllers/Http/CommentController.js:
```diff
@@ -0,0 +1,59 @@
+    return Comment.query()
+      .where((q) => {
+        q.orWhere('commenter', 'ilike', `%${query}%`)
+        q.orWhere('event_id', 'ilike', `%${query}%`)
```
I think these need to be different parameters.

`query` for comment text
`commenter_id` for `commenter_id` 
`event_id` for `event_id`

like

`?query=text&commenter=1&event_id=2`

lubien reviewed (changes requested):
You forgot to setup `routes.js`

How can I list replies? `/comments/id/replies`?

daniel-santana committed:
requested comments CRUD changes

lubien commented on app/Controllers/Http/CommentController.js:
```diff
@@ -4,12 +4,25 @@ const Comment = use('App/Models/Comment')
-        q.orWhere('commenter', 'ilike', `%${query}%`)
-        q.orWhere('event_id', 'ilike', `%${query}%`)
-      })
+      .where('answer_to', 'ilike', AnswerTo)
```
I don't think this should be a `like`

`.where('answer_to', answerTo)`

lubien commented on app/Controllers/Http/CommentController.js:
```diff
@@ -4,12 +4,25 @@ const Comment = use('App/Models/Comment')
+  }
+
+  answers({ request }) {
+    const { page = 1, AnswerTo } = request.get()
```
`answerTo` camelCase

lubien commented on app/Controllers/Http/CommentController.js:
```diff
@@ -4,12 +4,25 @@ const Comment = use('App/Models/Comment')
+    let query = Comment.query()
+
+    if (CommenterId) {
+      query = query.andWhere('commenter_id', 'ilike', CommenterId)
```
I don't think this should be a like

lubien commented on app/Controllers/Http/CommentController.js:
```diff
@@ -4,12 +4,25 @@ const Comment = use('App/Models/Comment')
+    }
+
+    if (EventId) {
+      query = query.andWhere('event_id', 'ilike', EventId)
```
I don't think this should be a like

lubien commented on app/Controllers/Http/CommentController.js:
```diff
@@ -4,12 +4,25 @@ const Comment = use('App/Models/Comment')
 class CommentController {
   index({ request }) {
-    const { page = 1, query = '' } = request.get()
+    const { page = 1, CommenterId, EventId } = request.get()
```
camelCase variables

lubien commented on database/migrations/1571807349075_comment_schema.js:
```diff
@@ -8,10 +8,17 @@ class CommentSchema extends Schema {
+        .references('users.id')
+        .onUpdate('CASCADE')
+        .onDelete('CASCADE')
       table.integer('event_id')
```
.unsigned()

lubien commented on database/migrations/1571807349075_comment_schema.js:
```diff
@@ -8,10 +8,17 @@ class CommentSchema extends Schema {
     this.create('comments', (table) => {
       table.increments()
       table.integer('commenter_id')
```
.unsigned()

daniel-santana committed:
added routes for event comments
corrected variable names (camel case)
corrected path name for comments (Model -> Models)
corrected comment table (unsigned foreign key, unecessary time column)

daniel-santana commented on app/Controllers/Http/CommentController.js:
```diff
@@ -0,0 +1,73 @@
+
+class CommentController {
+  index({ request }) {
+    const { page = 1, commenterId, eventId } = request.get()
```
commenterId and eventId are passed as params, not body

daniel-santana committed:
changed comments routes

lubien commented on app/Controllers/Http/CommentController.js:
```diff
@@ -0,0 +1,65 @@
+  async update({ request }) {
+    const data = request.all()
+
+    const { id } = request.params
```
Need to verify is user is owner or admin (see UserController)

lubien commented on app/Controllers/Http/CommentController.js:
```diff
@@ -0,0 +1,65 @@
+
+  async destroy({ request }) {
+    const { id } = request.params
+    const comment = await Comment.findOrFail(id)
```
Need to verify is user is owner or admin (see UserController)

lubien commented on app/Controllers/Http/CommentController.js:
```diff
@@ -0,0 +1,65 @@
+    const query = Comment.query()
+
+    if (replyTo) {
+      query.andWhere({ replyTo })
```
The column is named reply_to, does this work?

daniel-santana committed:
requested changes
verification if user is comment owner
